dldashboard/web/design_doc.md

```diff
@@ -0,0 +1,41 @@
+A high level design diagram of the current architecture of the dashboard is :
+
+![dashboard architecture diagram](https://craftassist.s3-us-west-2.amazonaws.com/pubr/backend+architecture.png)
+
+In the diagram above:
+- frontend: The React based frontend that hosts other React subcomponents.
+- StateManager: This module manages messaging between frontend and backend. We set the state of the frontend components through the data received socket.io connections between stateManager and backend. 
```

Comment:
A few questions that I think we should address here
* is the StateManager stateless (ie does it preserve its own state, if so how) or does it just receive the state from the backend?
* how does it communicate with the backend - polling, async calls? add API snippets for both sender and receiver?
* what is the frequency at which the frontend components's state is reset?

Reply:
> is the StateManager stateless (ie does it preserve its own state, if so how) or does it just receive the state from the backend?

The StateManager receives things from the backend  but also preserves state of frontend if any. The goal is to have it be completely in sync with backend but we aren't there yet.

> how does it communicate with the backend - polling, async calls? add API snippets for both sender and receiver?

socket io events.

> what is the frequency at which the frontend components's state is reset?
whenever there is a state change communicated from backend.